src/mixins/Mime.js

```diff
@@ -102,6 +102,14 @@ export default {
 		ext() {
 			return parsePath(this.basename).ext
 		},
+		queryString() {
+			const params = OC.Util.History.parseUrlQuery()
+			const dir = params.dir
+			delete params.dir
+			delete params.fileid
+			params.openfile = this.fileid
+			return 'dir=' + OC.encodePath(dir) + '&' + OC.buildQueryString(params)
```

Comment:
So what we are doing right now to create the new url is:
* extract the dir param
* remove the fileid param if there is one
* set the openfile param to the current file

I don't know what the talk url looks like. It probably does not have a dir param. So the resulting url will probably look like
```
previoustalkurl?dir=&openfile=2314&othertalkurlparamshere
```

Reply:
I checked... currently the url looks like this:
```
http://nextcloud.local/index.php/apps/spreed/?dir=undefined&openfile
```
The undefined is easy to get rid of by changing the order of the params. Not sure why `this.fileid` is not set.
I'll take a closer look at this later.

Reply:
I can already tell you, this won't scale. Any app can open the viewer, this is the whole purpose of it. So you cannot ensure every app will be compatible witht it.

Unless we add a dedicated way for apps to specify how they are managing the direct access to a resource, I'm against this as this is not standardized enough :)

And such implementation, will need to be super clean. :wink:

Reply:
@skjnldsv So what do you think would be the right take? Make this happen only if the viewer is opened in the files app - i.e. not running in standalone mode?

Reply:
No, others app might want it running in normal mode too.
I'm not really sure to be honest.
I was thinking about a specific optional parametter to give that is a function returning the direct link to the ressource.

I need to think abou tit a bit

Reply:
What if the viewer *app* does the history tracking and the *component* does not.

Because, as @skjnldsv said, when it's embedded in another application you might now want it to interfere with the app's history.

Reply:
I'll add my current thinking in the main thread of the discussion. Let's continue it there to make sure it stays even if the code i commented on here changes.